merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/AutoValueMappers.java

```diff
@@ -27,13 +31,14 @@
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 import java.util.regex.Pattern;
 
 public class AutoValueMappers {
-  private static final Pattern JAVA_IDENTIFIER_ILLEGAL_CHARACTERS = Pattern.compile("[.><,\\[\\]]");
+  private static final Pattern JAVA_IDENTIFIER_ILLEGAL_CHARACTERS = Pattern.compile("[^a-zA-Z0-9_]");
```

Comment:
The hope was to avoid the cat and mouse game of adding new illegal characters every time we encounter them. You're absolutely right that this whitelist is much too restrictive.

Looks like there are two methods we can use: `Character.isJavaIdentifierStart` and `Character.isJavaIdentifierPart`. I think this method will be sufficiently future-proof:
```java
  /**
   * Turn a given string into a valid Java identifier
   * @param str a string that may or may not be a valid Java identifier
   * @return a string that is a valid Java identifier
   */
  public static String getIdentifier(String str) {
    if (str.length() == 0) throw new IllegalArgumentException("Cannot turn empty string into an identifier");
    final var identifier = new StringBuilder();
    if (Character.isJavaIdentifierStart(str.charAt(0))) {
      identifier.append(str.charAt(0));
    } else {
      identifier.append("_");
    }
    for (int i = 1; i < str.length(); i++) {
      if (Character.isJavaIdentifierPart(str.charAt(i))) {
        identifier.append(str.charAt(i));
      } else {
        identifier.append("_");
      }
    }
    return identifier.toString();
  }
```

Reply:
I agree. Adjacent topic that's out of scope for this PR: Can we use this method where we interpret the activity type name in `@ActivityType`? IIRR we're currently just assuming that they're valid Java/TS identifiers, and parts of Aerie break if that assumption is false (ie Constraints).

Reply:
Made a ticket for this: https://github.com/NASA-AMMOS/aerie/issues/1223